### PR TITLE
Fixed XML singular/pluralization

### PR DESCRIPTION
**Pull request by herariom**
#62 XML elements should reflect their proper pluralization.
Used suggested Pluralize framework to fix XML element pluralization

- Elements will be pluralized if they have child elements
- Child elements will be singularized
- Elements with no children will be singularized

### CODE REVIEW

File: FakeServer/Common/Formatters/XmlOutputFormatter.cs [hunk 1]
```diff
@@ -24,20 +25,19 @@ public XmlOutputFormatter()
         protected override bool CanWriteType(Type type) =>
                 typeof(ExpandoObject).IsAssignableFrom(type) || typeof(IEnumerable<object>).IsAssignableFrom(type) ? base.CanWriteType(type) : false;
 
+        private static Pluralizer pluralizer = new Pluralizer();
```
Comment (ttu) [inline FakeServer/Common/Formatters/XmlOutputFormatter.cs]: This project uses underscore naming convention for private fields (or should use, there might be some places that don't follow this).

```cs
private static Pluralizer _pluralizer = new Pluralizer();
````

File: FakeServer/Common/Formatters/XmlOutputFormatter.cs [hunk 2]
```diff
@@ -24,20 +25,19 @@ public XmlOutputFormatter()
...
             XElement HandleExpandoField(XElement acc, KeyValuePair<string, object> fields)
             {
                 XElement element = null;
 
+
                 if (fields.Value is IEnumerable<object> innerList)
                 {
-                    var children = innerList.Select((i, idx) => ((ExpandoObject)i).Aggregate(new XElement($"{fields.Key}_{idx}"), HandleExpandoField));
-                    element = new XElement(fields.Key, children);
+                    var children = innerList.Select((i, idx) => ((ExpandoObject)i).Aggregate(new XElement($"{pluralizer.Singularize(fields.Key)}_{idx}"), HandleExpandoField));
```
Comment (ttu) [inline FakeServer/Common/Formatters/XmlOutputFormatter.cs]: Now idx can be removed.

File: FakeServer/Common/Formatters/XmlOutputFormatter.cs [hunk 3]
```diff
@@ -52,8 +52,8 @@ XElement HandleExpandoField(XElement acc, KeyValuePair<string, object> fields)
 
             XElement MultipleItemsToXml(string name, IEnumerable<object> itemCollection)
             {
-                var children = itemCollection.Select((i, idx) => ((ExpandoObject)i).Aggregate(new XElement($"{name}_{idx}"), HandleExpandoField));
-                var root = new XElement(name, children);
+                var children = itemCollection.Select((i, idx) => ((ExpandoObject)i).Aggregate(new XElement($"{pluralizer.Singularize(name)}_{idx}"), HandleExpandoField));
```
Comment (ttu) [inline FakeServer/Common/Formatters/XmlOutputFormatter.cs]: Idx can be removed also from here

**Comment by ttu** [comment]
Thanks for the implementation! Just few minor review change requests.

Could you also add few asserts to the XML test for element naming? Something like this

```cs
        [Fact]
        public async Task GetSingleFamily_Accept_XML()
        {
            var request = new HttpRequestMessage(new HttpMethod("GET"), $"api/families/1");
            request.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue("text/xml"));

            var result = await _fixture.Client.SendAsync(request);
            result.EnsureSuccessStatusCode();

            var xml = await result.Content.ReadAsStringAsync();
            Assert.False(string.IsNullOrEmpty(xml));

            // Check that inner collections are serialized correctly
            Assert.DoesNotContain("[id, 0]", xml);
            Assert.DoesNotContain("System.Collections.Generic.List", xml);

            var rows = xml.Split(Environment.NewLine);
            Assert.Equal("<parents>", rows[3].Trim());
            Assert.Equal("<parent>", rows[4].Trim());
            Assert.Equal("<friends>", rows[44].Trim());
            Assert.Equal("<friend>", rows[45].Trim());
        }
```

**Comment by herariom** [comment]
Added changes as requested

**Comment by ttu** [comment]
Thanks for the fix 👍

**Pull request merged**